Tweak wheel scaling behavior to be more consistent

AlexTMjugador opened the pull request:
#1641, which introduced `setVehicleWheelScale`, doesn't account for GTA: SA changing the wheel scale as a result of wheel upgrades being added or removed, and using different default scales for vehicles with upgraded wheels. Although this does not pose a serious issue, I have just found that it leads to inconsistent behavior, like the one that is shown in the following flow of events:


> - Wheel upgrade is removed from streamed in vehicle
> - Wheel scale is reset by GTA to the default, which is returned by getVehicleWheelScale
> - Vehicle is streamed out and back in
> - The wheel scale is set back to the last value set by a script or 1 (not the same scale that it was streamed out with)


This PR tweaks that behavior by adding logic to synchronize the default wheel scale calculated by GTA with the one present in `CClientVehicle` attributes, for which functionality added in #1644 is used. This way, the previous event flow, for example, becomes:


> - Wheel upgrade is removed from streamed in vehicle
> - Wheel scale is reset by GTA
> - Vehicle is streamed out and back in
> - The wheel scale set by GTA is preserved


Also, as expected, the scale set by a script is now always honored, until it is reset by GTA during a script-initiated wheel upgrade installation or removal.

All in all, I believe that the new behavior is more consistent with the expectations that script developers may have, and as the function was introduced only days ago, it's not too late to fix this. Moreover, I think that GTA resetting the wheel size makes sense because, as upgraded wheel models are reused accross different vehicles, they need different scales to accomodate each vehicle, so it should be changed anyway. Finally, as a side bonus, the new behavior works better for streamed out vehicles.

AlexTMjugador committed:
Tweak wheel scaling behavior to be more consistent

AlexTMjugador committed:
Handle m_pUpgrades being nullptr in CClientVehicle::ResetWheelScale

m_pUpgrades shouldn't be null unless its allocation during the
constructor fails without throwing an exception, but other functions
in the same file check for null before using m_pUpgrades, so add a
check here too to be safe.

AlexTMjugador committed:
Revert "Handle m_pUpgrades being nullptr in CClientVehicle::ResetWheelScale"

This reverts commit b0880ad7b1a78e01226516750c009cd832879a9e.

Checking for nullptr here actually is probably overly defensive, as the
new operator that allocates this field may throw an exception. I think
it is better to be a bit offensive here, because if m_pUpgrades becomes
null something is going very wrong.

AlexTMjugador committed:
Assert that m_pUpgrades is not null in CClientVehicle::ResetWheelScale

As explained in previous commits, if the software behaves as specified,
checking for m_pUpgrades is not necessary. Instead, check that invariant
is being held via an assertion, which only impacts debug builds. This
may be useful to catch errors if the `CClientVehicle` class is extended
in such a way that a subclass sets m_pUpgrades to nullptr.

qaisjp commented:
> as the function was introduced only days ago, it's not too late to fix this.

Don't worry! I _think_ stuff that we introduce into master can be changed without worry for BC until it is actually released in a numbered version.

qaisjp commented on Client/mods/deathmatch/logic/CClientVehicle.cpp:
```diff
@@ -2825,10 +2827,21 @@ void CClientVehicle::Create()
+
         // Re-add all the upgrades - Has to be applied after handling *shrugs*
+        bool bPreviousWheelScaleChanged = m_bWheelScaleChanged;
+        float fPreviousWheelScale = m_fWheelScale;
```
```suggestion
        // Applying upgrades can change these values,
        // so we should keep track of the original values.
        bool bPreviousWheelScaleChanged = m_bWheelScaleChanged;
        float fPreviousWheelScale = m_fWheelScale;

        // Re-add all the upgrades - Has to be applied after handling *shrugs*
```

AlexTMjugador replied:
Thank you! I've just commited a variant of your suggested change, changing the comments a bit to make them more clear and concise.

qaisjp reviewed (approved):
I've left a nit, where primarily `// Re-add all the upgrades - Has to be applied after handling *shrugs*` didn't make sense above these two assignments.

This LGTM. This sounds super reasonable, thank you very much for the explanation, and thank you for continuously testing these functions 🎉

AlexTMjugador committed:
Address suggested change by @qaisjp

AlexTMjugador committed:
Restore m_fWheelScale in CClientVehicle::Create if needed

Restoring m_fWheelScale to fPreviousWheelScale is needed to handle
succesive restreams properly.
